### PR TITLE
capz/conformance: update pull capz conformance job

### PR DESCRIPTION
**Pull request by cpanato**
update conformance job to support: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/986

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml [hunk 1]
```diff
@@ -171,45 +171,38 @@ presubmits:
...
       preset-kind-volume-mounts: "true"
-      preset-azure-cred: "true"
-    path_alias: "sigs.k8s.io/cluster-api-provider-azure"
-    extra_refs:
-    - org: kubernetes
-      repo: kubernetes
-      base_ref: v1.18.6
-      path_alias: k8s.io/kubernetes
+      preset-azure-cred-only: "true"
+      preset-azure-anonymous-pull: "true"
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml:185]: do we need the ACR pull here?

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml [hunk 2]
```diff
@@ -171,45 +171,38 @@ presubmits:
...
-            requests:
-              cpu: 1
-              memory: "4Gi"
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20201005-0b243c0-1.18
+        command:
+        - "runner.sh"
+        - "./scripts/ci-conformance.sh"
+        env:
+          - name: GINKGO_FOCUS
+            value: "Cluster API E2E tests"
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml:194]: I don't think that's the right FOCUS, should be `Conformance Tests`?

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml [hunk 3]
```diff
@@ -171,45 +171,38 @@ presubmits:
...
+          privileged: true
+        resources:
+          requests:
+            cpu: 2
+            memory: "9Gi"
     annotations:
       testgrid-dashboards: sig-cluster-lifecycle-cluster-api-provider-azure
       testgrid-tab-name: pull-conformance-v1alpha3
+      testgrid-alert-email: k8s-infra-staging-cluster-api-azure@kubernetes.io
   - name: pull-cluster-api-provider-azure-conformance-machine-pool-v1alpha3
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml:206]: let's also update the machine pool flavor conformance. Unless we're able to remove it altogether and have two templates run conformance in parallel

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml (hunk 1 above)
Comment (cpanato) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml:185]: It pushes the images to the registry, do you think we don't need that?

**Review by CecileRobertMichon — APPROVED** [review]
/lgtm
/approve

**Pull request merged**